plugin/helpers_bots.go

```diff
@@ -4,12 +4,84 @@
 package plugin
 
 import (
+	"io/ioutil"
+	"path/filepath"
+
 	"github.com/mattermost/mattermost-server/model"
 	"github.com/mattermost/mattermost-server/utils"
 	"github.com/pkg/errors"
 )
 
-func (p *HelpersImpl) EnsureBot(bot *model.Bot) (retBotId string, retErr error) {
+type ensureBotOptions struct {
+	ProfileImagePath string
+	IconImagePath    string
+}
+
+type EnsureBotOption func(*ensureBotOptions)
```

Comment:
Please don't export this
```suggestion
type ensureBotOption func(*ensureBotOptions)
```

Reply:
@hanzei I won't be able to pass in options in the unit tests if I don't export it (making any additional tests a bit redundant)

Reply:
You can use `ProfileImagePath` and `IconImagePath` in your unit test. IMO a unit test should focus on the public API and this is these two methods in this case.